File: app/src/main/java/za/org/grassroot2/services/UserDetailsService.java
package za.org.grassroot2.services;

import android.accounts.Account;

import io.reactivex.Single;
import za.org.grassroot2.model.UserProfile;

/**
 * Created by luke on 2017/08/09.
 */

public interface UserDetailsService extends GrassrootService {

    Single<UserProfile> storeUserDetails(final String userUid,
                                         final String userPhone,
                                         final String userDisplayName,
                                         final String userEmailAddress,
                                         final String userLanguageCode,
                                         final String userSystemRole,
                                         final String userToken);

    Account setAuthToken(String userToken);

    Single<Boolean> logout(boolean deleteAndroidAccount, boolean wipeRealm);

    boolean isSyncCompleted();

    boolean isSyncFailed();

    void setSyncState(int status);

    String getCurrentToken();
    String getCurrentUserUid();
    String getCurrentUserMsisdn();

    void requestSync();

}

File: app/src/main/java/za/org/grassroot2/services/UserDetailsService.kt
package za.org.grassroot2.services

import android.accounts.Account

import io.reactivex.Single
import za.org.grassroot2.model.UserProfile

/**
 * Created by luke on 2017/08/09.
 */

interface UserDetailsService {

    val isSyncCompleted: Boolean

    val isSyncFailed: Boolean

    val currentToken: String?
    val currentUserUid: String?
    val currentUserMsisdn: String?

    fun storeUserDetails(userUid: String,
                         userPhone: String,
                         userDisplayName: String,
                         userEmailAddress: String?,
                         userLanguageCode: String,
                         userSystemRole: String?,
                         userToken: String): Single<UserProfile>

    fun setAuthToken(userToken: String): Account

    fun logout(deleteAndroidAccount: Boolean, wipeDb: Boolean): Single<Boolean>

    fun setSyncState(status: Int)

    fun requestSync()

}